### ISSUE TITLE
ApplePicker Error in upstream software stack

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#217]
In #181 , I cleaned up existing warnings and also checked upstream warnings (those with newer versions of dependencies than we currently employ).  Upon letting python take the latest packages it saw fit, I was greeted with the following error:

`ApplePickerTestCase.testPickCenters failed with AssertionError: (1321.0, 361.0) not an expected center.`

I think we'd need to dig into that.

Generally, people seem satisfied with the function of ApplePicker.  There may be one or two small ApplePicker issues around, and so it might make sense to have an ApplePicker development iteration later after other priorities are addressed. We can stand up a project board for those issues at that time.

**Comment by garrettwrong** [comment]
Generally I saw this on Linux, but the fact it reproduces same on OSX is actually a good thing, should be easier to suss out. 
 
Notes on reproducing this on OSX (10.14.6):

```
# In the root of an ASPIRE repo, get the latest code so we're looking at same thing...
git fetch origin
git checkout origin/develop

# Create a fresh environment
conda env create -f environment.yml --name aspire_test_issue217
conda activate aspire_test_issue217
# Use this directory as your "install", ie, developer mode
pip install -e .

# Run the test, should pass, success.
pytest -s tests/test_apple.py

# Below is one hacky way to upgrade all your packages
# You will see red stuff like this:
#   aspire 0.5.28 requires numpy==1.16, but you'll have numpy 1.19.1 which is incompatible.
# This is because we "fix" the versions in setup.py.
#   We are intentionally upgrading for testing this issue, so we ignore the noise.
#
pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1  | xargs -n1 pip install -U


# Run the test again, should fail. Bummer.
pytest -s tests/test_apple.py
```

The actual traceback is long, so I included these notes, the traceback, and list of upgraded package versions in the attached file.
[ay.txt](https://github.com/ComputationalCryoEM/ASPIRE-Python/files/5042313/ay.txt)


As an aside, two out of the six times I ran this test to make these notes above I randomly got the segfault from #150 #216 (once old pckages, once new). If you happen to see it too at any point, please note in one of those places. I'm just curious.

**Comment by garrettwrong** [comment]
This should have been closed by #221 .Closing now.
